### ISSUE TITLE
git diffs outputs "File: *** is outside repository" on Update Gutter

### ISSUE THREAD

**Issue by haidran** [textmate/scm-diff-gutter.tmbundle#9]
I'm having an issue with this bundle: nothing appears in the gutter even when changes are made to a file in a git repository.

Using aborts to debug the *Update Gutter* command, I've found that providing the absolute path to the file when invoking `git diff` gives a "**File: *** is outside repository**". If I recreate the executed command in Terminal, the same happens, even if I ```cd``` to the repository directory. Using a relative path, the diff works as expected.

I have used system's git and Homebrew's git for the tests, and this problem persists. Hardcoding the relative path to the file in ```diff_cmd```, the gutter shows diff as expected. I'm using High Sierra 10.13.1.

**Comment by sorbits** [comment]
> TextMate needs to feed its paths (TM_FILEPATH for example) to git respecting the path original normalization.

A little hard to wrap my head around this issue.

It would be wrong for TextMate to start to apply normalization transformations on strings returned from the file system APIs (which is what gets used with `TM_FILEPATH` etc.).

A normalization transformation would only be proper if e.g. the user enters a name (say, in a dialog) and TextMate then creates a file using that string (here it could normalize to whatever format Apple think their file system names should use).

I currently don’t have APFS, but I can test this when I do have it installed.

For the records, are you using the case-insensitive or case-sensitive version?

**Comment by haidran** [comment]
I'm using the case-insensitive variant of APFS.

After some tests and I huge load of patience, I've determined that this may be a **git** or a **file system** issue after all.

Using relative paths with UTF-8 special characters work pretty well when executing "git diff". Using absolute paths, strange things happen, in a non-reproducible way.

With this I mean: changing directory to the root of a git project located in a path that includes special characters and doing **git diff "example.txt"** and **git diff "$(pwd)/example.txt"** doesn't give the same output. The first command works as expected, the second fails: "the specified path is outside directory". This same issue doesn't happen in other similar cases, and I haven't found a logical reason behind this yet. What I'm sure of is this is related, in some way, to APFS, as HFS+ didn't give my any problem with this kind of things.

Summing up, TextMate is working like it should, this issues are related to git or the file system, I'm not quite sure. Sorry to bother you with this, and thank you and all contributors for this awesome editor 😃
